backend/routes/query.py

```
@@ -0,0 +1,91 @@
+from ..dependencies import get_services
+from ..models.request_models import Query
+from ..security.validator import SecurityValidator
+
+# Initialize router with rate limiting
+limiter = Limiter(key_func=get_remote_address)
+router = APIRouter()
+
+
+@router.post("/query")
+@limiter.limit(AppConfig.RATE_LIMIT)
+async def query_endpoint(request: Request, query: Query, services: dict = Depends(get_services)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)